operator/k8s/resource_ctors.go

```diff
@@ -94,3 +95,23 @@ func CiliumBGPNodeConfigOverrideResource(lc cell.Lifecycle, cs client.Clientset,
 	)
 	return resource.New[*cilium_api_v2alpha1.CiliumBGPNodeConfigOverride](lc, lw, resource.WithMetric("CiliumBGPNodeConfigOverride")), nil
 }
+
+func PodResource(lc cell.Lifecycle, cs client.Clientset, opts ...func(*metav1.ListOptions)) (resource.Resource[*slim_corev1.Pod], error) {
+	if !cs.IsEnabled() {
+		return nil, nil
+	}
+	lw := utils.ListerWatcherWithModifiers(
+		utils.ListerWatcherFromTyped[*slim_corev1.PodList](cs.Slim().CoreV1().Pods("")),
+		opts...,
+	)
+
+	indexers := cache.Indexers{
```

Comment:
Maybe add a comment that the indexer is used for operator only when it's managing CIDs, because it needs to list the pods by namespace in case namespace labels change, so it doesn't have to list all pods and match namespace.

Reply:
I've added the details in the commit description, but I'm happy to add it as a comment too